applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1693,6 +1777,99 @@ void BaseSolidElement::CalculateAndAddExtForceContribution(
 /***********************************************************************************/
 /***********************************************************************************/
 
+void BaseSolidElement::CalculateLumpedMassVector(VectorType& rMassVector)
+{
+    KRATOS_TRY;
+
+    const auto& r_geom = GetGeometry();
+    const auto& r_prop = GetProperties();
+    const SizeType dimension = r_geom.WorkingSpaceDimension();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType mat_size = dimension * number_of_nodes;
+
+    // Clear matrix
+    if (rMassVector.size() != mat_size)
+        rMassVector.resize( mat_size, mat_size );
+
+    const double density = r_prop[DENSITY];
+    const double thickness = (dimension == 2 && r_prop.Has(THICKNESS)) ? r_prop[THICKNESS] : 1.0;
+
+    // LUMPED MASS MATRIX
+    const double total_mass = GetGeometry().DomainSize() * density * thickness;
+
+    Vector lumping_factors;
+    lumping_factors = GetGeometry().LumpingFactors( lumping_factors );
+
+    for ( IndexType i = 0; i < number_of_nodes; ++i ) {
+        const double temp = lumping_factors[i] * total_mass;
+        for ( IndexType j = 0; j < dimension; ++j ) {
+            IndexType index = i * dimension + j;
+            rMassVector[index] = temp;
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseSolidElement::CalculateLumpedDampingMatrix(
+    MatrixType& rDampingMatrix,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    unsigned int number_of_nodes = GetGeometry().size();
```

Comment:
This function is protected, so why do you give back a matrix instead of a vector? (you can define the interface!)

aside of this you can avoid computing anything in the case in which the rayleigh coefficients are zero.

Reply:
The damping matrix is not lumped, I will rename the method. The only terms are diagonal are the relative to the mass, not the ones relative to the stiffness

Reply:
so its not possible to use the current method `CalculateDampingMatrix` with the regular StiffnessMatrix and the Lumped-MassMatrix?

Reply:
It is indeed what is done

Reply:
ah and the only difference is that you use an optimized version of calculating the lumped mass-matrix  here right?